pkg/scheduler/scheduler_test.go

```diff
@@ -2532,119 +2533,119 @@ func TestResourcesToReserve(t *testing.T) {
 		name            string
 		assignmentMode  flavorassigner.FlavorAssignmentMode
 		borrowing       bool
-		assignmentUsage cache.FlavorResourceQuantities
-		cqUsage         cache.FlavorResourceQuantities
-		wantReserved    cache.FlavorResourceQuantities
+		assignmentUsage resources.FlavorResourceQuantities
+		cqUsage         resources.FlavorResourceQuantities
+		wantReserved    resources.FlavorResourceQuantities
 	}{
 		{
 			name:           "Reserved memory and gpu less than assignment usage, assignment preempts",
 			assignmentMode: flavorassigner.Preempt,
-			assignmentUsage: cache.FlavorResourceQuantities{
-				kueue.ResourceFlavorReference("on-demand"): {corev1.ResourceMemory: 50},
-				kueue.ResourceFlavorReference("model-a"):   {"gpu": 6},
-			},
-			cqUsage: cache.FlavorResourceQuantities{
-				kueue.ResourceFlavorReference("on-demand"): {corev1.ResourceMemory: 60},
-				kueue.ResourceFlavorReference("spot"):      {corev1.ResourceMemory: 50},
-				kueue.ResourceFlavorReference("model-a"):   {"gpu": 6},
-				kueue.ResourceFlavorReference("model-b"):   {"gpu": 2},
-			},
-			wantReserved: cache.FlavorResourceQuantities{
-				kueue.ResourceFlavorReference("on-demand"): {corev1.ResourceMemory: 40},
-				kueue.ResourceFlavorReference("model-a"):   {"gpu": 4},
-			},
+			assignmentUsage: resources.FlavorResourceQuantitiesFlat{
+				resources.FlavorResource{Flavor: kueue.ResourceFlavorReference("on-demand"), Resource: corev1.ResourceMemory}: 50,
```

Comment:
```suggestion
				{Flavor: kueue.ResourceFlavorReference("on-demand"), Resource: corev1.ResourceMemory}: 50,
```

check if this is possible